diff --git a/src/Plugin/Filter/RelToAbs.php b/src/Plugin/Filter/RelToAbs.php
index a8f56b6..0769003 100644
--- a/src/Plugin/Filter/RelToAbs.php
+++ b/src/Plugin/Filter/RelToAbs.php
@@ -68,7 +68,7 @@ class RelToAbs extends FilterBase {
         else {
           $new_txt .= substr($txt, 0, $pos);
         }
-        $txt = substr($txt, $pos);
+        $txt = ltrim(substr($txt, $pos), '/');
         // Remove leading sub-directory prefix if site installed in subdir.
         if ($sub_dir) {
           if (substr($txt, 0, strlen($sub_dir)) == $sub_dir) {
